### PR TITLE
Simulation build infrastructure and CMake support

### PR DESCRIPTION
**Pull request by ljanyst**
This CL makes it possible to build self-contained "simulation apps" bundling all the SystemVerilog and DPI code and run them using runtime scripts for either Xsim or Modelsim. It also includes the CMake build infrastructure to declare and build SystemVerilog libraries and link them against SystemVerilog apps. All the dependency resolution works as well, so that, if any of the files required to build the libraries/apps has been changed, all the necessary targets will be seemlessly rebuilt.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+endif()
+
+include(SystemVerilog)
+
+add_subdirectory(src)
+add_subdirectory(tests)
+
+message(STATUS "----------------------------------------" )
+message(STATUS "Use Xsim:     " ${USE_XSIM} )
+message(STATUS "USE Modelsim: " ${USE_MODELSIM} )
```
Comment (jfrohnhofen) [inline CMakeLists.txt]: Nit: Different capitalization of "use"

**Review by jfrohnhofen — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+
+sw-axi
+======
+
+Building and running instructions
+---------------------------------
+
+`sw-axi` uses CMake for building. It requires the folloing packages on Debian/Ubuntu:
```
Comment (Richard-Gordon) [inline README.md]: Typo: Following

File: README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+ * `build-essential`
+ * `cmake`
+ * `libflatbuffers-dev`
+ * `golang`
+
+It also requires one of the folowing simulators:
```
Comment (Richard-Gordon) [inline README.md]: Typo: Following

**Review by Richard-Gordon — APPROVED** [review]

**Comment by ljanyst** [comment]
Thanks! Fixed!

**Pull request merged**